compose-custom-attachments/src/main/java/io/getstream/chat/android/compose/customattachments/ui/QuotedDateAttachmentFactory.kt

```diff
@@ -0,0 +1,86 @@
+package io.getstream.chat.android.compose.customattachments.ui
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import io.getstream.chat.android.compose.customattachments.R
+import io.getstream.chat.android.compose.state.messages.attachments.AttachmentState
+import io.getstream.chat.android.compose.ui.attachments.AttachmentFactory
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+
+/**
+ * A custom [AttachmentFactory] that adds support for quoted date attachments.
+ */
+val quotedDateAttachmentFactory: AttachmentFactory = AttachmentFactory(
+    canHandle = { attachments -> attachments.any { it.type == "date" } },
+    content = @Composable { modifier, attachmentState ->
+        QuotedDateAttachmentContent(
+            modifier = modifier,
+            attachmentState = attachmentState
+        )
+    }
+)
+
+/**
+ * Represents the UI shown in the message list for quoted date attachments.
+ *
+ * @param attachmentState The state of the attachment.
+ * @param modifier Modifier for styling.
+ */
+@Composable
+fun QuotedDateAttachmentContent(
+    attachmentState: AttachmentState,
+    modifier: Modifier = Modifier,
+) {
+    val attachment = attachmentState.message.attachments.first { it.type == "date" }
+    val date = attachment.extraData["payload"].toString()
+    val formattedDate = StringBuilder().apply {
+        val dateTime = SimpleDateFormat("MMMMM dd, yyyy", Locale.getDefault()).parse(date) ?: return@apply
```

Comment:
Not sure if `parse` will even return `null`. I think it either returns a date or an error.

Reply:
I will add a try catch around it and if there is an exception will handle it. Also compiler is complaining that Date is nullable since we are calling a java function so i will check for null just not to have the warning.